interfaces/builtin/utils_test.go

```diff
@@ -65,6 +65,84 @@ func (s *utilsSuite) TestImplicitSystemConnectedSlot(c *C) {
 	c.Assert(builtin.ImplicitSystemConnectedSlot(s.conSlotSnapd), Equals, true)
 }
 
+const yaml = `name: test-snap
+version: 1
+plugs:
+ x11:
+slots:
+ opengl:
+apps:
+ app1:
+  command: bin/test1
+  plugs: [home]
+  slots: [unity8]
+ app2:
+  command: bin/test2
+  plugs: [home]
+hooks:
+ install:
+  plugs: [network,network-manager]
+ post-refresh:
+  plugs: [network,network-manager]
+`
+
+func (s *utilsSuite) TestLabelExpr(c *C) {
+	info := snaptest.MockInfo(c, yaml, nil)
+
+	// all apps and all hooks
+	label := builtin.LabelExpr(info.Apps, info.Hooks, info)
+	c.Check(label, Equals, `"snap.test-snap.*"`)
+
+	// all apps, no hooks
+	label = builtin.LabelExpr(info.Apps, nil, info)
+	c.Check(label, Equals, `"snap.test-snap.{app1,app2}"`)
+
+	// one app, no hooks
+	label = builtin.LabelExpr(map[string]*snap.AppInfo{"app1": info.Apps["app1"]}, nil, info)
+	c.Check(label, Equals, `"snap.test-snap.app1"`)
+
+	// no apps, one hook
+	label = builtin.LabelExpr(nil, map[string]*snap.HookInfo{"install": info.Hooks["install"]}, info)
+	c.Check(label, Equals, `"snap.test-snap.hook.install"`)
+
+	// one app, all hooks
+	label = builtin.LabelExpr(map[string]*snap.AppInfo{"app1": info.Apps["app1"]}, info.Hooks, info)
+	c.Check(label, Equals, `"snap.test-snap.{app1,hook.install,hook.post-refresh}"`)
+
+	// only hooks
+	label = builtin.LabelExpr(nil, info.Hooks, info)
+	c.Check(label, Equals, `"snap.test-snap.{hook.install,hook.post-refresh}"`)
+
+	// nothing
+	label = builtin.LabelExpr(nil, nil, info)
```

Comment:
I wonder if in this case we could reduce it to `snap.core.*` as it's still true that it selects all (none) apps and hooks. Opinions?

Reply:
Yeah, I was considering that but wasn't sure if it's not going to be confusing (as meaning "all"). I'm fine to change this if it's deemed fine.

Reply:
I prefer having something that matches nothing if there is nothing to match, that something that matches anything.